[KAFKA-6702]Wrong className in LoggerFactory.getLogger method

hejiefang opened the pull request:
[https://issues.apache.org/jira/browse/KAFKA-6702](https://issues.apache.org/jira/browse/KAFKA-6702)
Wrong className in LoggerFactory.getLogger method

Commit:
[KAFKA-6702]Wrong className in LoggerFactory.getLogger method

Commit:
[KAFKA-6702]Wrong className in LoggerFactory.getLogger method

omkreddy commented on streams/src/main/java/org/apache/kafka/streams/errors/LogAndContinueExceptionHandler.java:
```diff
@@ -30,7 +30,7 @@
  */
 public class LogAndContinueExceptionHandler implements DeserializationExceptionHandler {
-    private static final Logger log = LoggerFactory.getLogger(StreamThread.class);
+    private static final Logger log = LoggerFactory.getLogger(LogAndContinueExceptionHandler.class);
```
Pls fix checkstyle error due to unused StreamThread import

hejiefang replied:
I fix that，but another error appear. If there is any exception in test of testNoConsumeWithoutDescribeAclViaSubscribe, i have only change the class name of logger and do nothing for zookeeper.set.acl

omkreddy replied:
that may be transient error.. we can re-run the tests again by commenting "retest this please"

hejiefang replied:
thanks

hejiefang replied:
Execuse me,is there any transient error in the test. I can not see the result now. It says 0 failed but JDK 7 and Scala 2.11 is FAILURE

omkreddy commented:
retest this please

rajinisivaram committed:
MINOR: Fix AdminClient.describeConfigs() of listener configs (#4747)

Don't return config values from `describeConfigs` if the config type cannot be determined. Obtain config types correctly for listener configs for `describeConfigs` and password encryption.

Reviewers: Jason Gustafson <jason@confluent.io>

perkss committed:
KAFKA-6659: Improve error message if state store is not found (#4732)

Reviewers: Bill Bejeck <bill@confluent.io>, Matthias J. Sax <matthias@confluent.io>

bbejeck committed:
MINOR: Rolling bounce upgrade fixed broker system test (#4690)

Reviewers: Guozhang Wang <guozhang@confluent.io>, John Roesler <john@confluent.io>, Matthias J. Sax <matthias@confluent.io>

guozhangwang committed:
KAFKA-6611: PART I, Use JMXTool in SimpleBenchmark (#4650)

1. Use JmxMixin for SimpleBenchmark (will remove the self reporting in #4744), only when loading phase is false (i.e. we are in fact starting the streams app).

2. Reported the full jmx reported metrics in log files, and in the returned data only return the max values: this is because we want to skip the warming up and cooling down periods that will have lower rate numbers, while max represents the actual rate at full speed.

3. Incorporates two other improves to JMXTool: #1241 and #2950

Reviewers: John Roesler <john@confluent.io>, Matthias J. Sax <matthias@confluent.io>, Rohan Desai <desai.p.rohan@gmail.com>

hachikuji committed:
KAFKA-6683; Ensure producer state not mutated prior to append (#4755)

We were unintentionally mutating the cached queue of batches prior to appending to the log. This could have several bad consequences if the append ultimately failed or was truncated. In the reporter's case, it caused the snapshot to be invalid after a segment roll. The snapshot contained producer state at offsets higher than the snapshot offset. If we ever had to load from that snapshot, the state was left inconsistent, which led to an error that ultimately crashed the replica fetcher.

The fix required some refactoring to avoid sharing the same underlying queue inside ProducerAppendInfo. I have added test cases which reproduce the invalid snapshot state. I have also made an effort to clean up logging since it was not easy to track this problem down.

One final note: I have removed the duplicate check inside ProducerStateManager since it was both redundant and incorrect. The redundancy was in the checking of the cached batches: we already check these in Log.analyzeAndValidateProducerState. The incorrectness was the handling of sequence number overflow: we were only handling one very specific case of overflow, but others would have resulted in an invalid assertion. Instead, we now throw OutOfOrderSequenceException.

Reviewers: Apurva Mehta <apurva@confluent.io>, Jun Rao <junrao@gmail.com>

ijuma committed:
MINOR: Java 9/10 fixes, gradle and minor deps update (#4725)

* Added dependencies so that Trogdor and Connect work with Java 9 and 10
* Updated Jacoco to 0.8.1 so that it works with Java 10
* Updated Gradle to 4.6
* A few minor version bumps (not related to Java9/10 fixes)

I tested manually that we can run ./gradlew test with Java 10
after these changes. There are test failures as EasyMock
and PowerMock will have to be updated to use a newer
ASM version. But compiling successfully and most tests
passing is progress. :)

I also tested manually that Trogdor can be started with Java 10.
It previously failed with a ClassNotFoundError.

Reviewers: Jason Gustafson <jason@confluent.io>

rajinisivaram committed:
MINOR: Fix encoder config to make DynamicBrokerReconfigurationTest stable (#4764)

DynamicBrokerReconfigurationTest currently assumes that passwords encoded with one secret will fail with an exception if decoded with another secret and configures an old.secret in setUp. This could potentially cause test failures if a password was incorrectly decoded with the wrong secret, since the test writes passwords encoded with the new secret directly to ZooKeeper. Since old.secret is only used in one test for verifying secret rotation, this config can be moved to that test to avoid transient failures.

Reviewers: Ismael Juma <ismael@juma.me.uk>

gitlw committed:
KAFKA-6612: Added logic to prevent increasing partition counts during topic deletion

This patch adds logic in handling the PartitionModifications event, so that if the partition count is increased when a topic deletion is still in progress, the controller will restore the data of the path /brokers/topics/"topic" to remove the added partitions.

Testing done:
Added a new test method to cover the bug

Author: Lucas Wang <luwang@linkedin.com>

Reviewers: Jiangjie (Becket) Qin <becket.qin@gmail.com>

Closes #4666 from gitlw/prevent_increasing_partition_count_during_topic_deletion

hachikuji commented:
Note `LogAndFailExceptionHandler` has the same issue. Maybe we should fix it as well?

huxi-2b committed:
KAFKA-6637; Avoid divide by zero error with segment.ms set to zero (#4698)

Require a minimum value of 1 for `segment.ms` to avoid division by zero when computing random jitter.

Reviewers: Manikumar Reddy O <manikumar.reddy@gmail.com>, Jason Gustafson <jason@confluent.io>

asasvari committed:
MINOR: Fix potential resource leak in FileOffsetBackingStore (#4739)

Reviewers: Sandor Murakozi <smurakozi@gmail.com>, Jason Gustafson <jason@confluent.io>

cburroughs committed:
MINOR: Remove redundant initialization of `Stats.index` (#4751)

rajinisivaram committed:
KAFKA-6710: Remove Thread.sleep from LogManager.deleteLogs (#4771)

`Thread.sleep` in `LogManager.deleteLogs` potentially blocks a scheduler thread for up to `log.segment.delete.delay.ms` with a default value of a minute. To avoid this, `deleteLogs` now deletes the logs for which `currentDefaultConfig.fileDeleteDelayMs` has elapsed after the delete was scheduled. Logs for which this interval has not yet elapsed are considered for deletion in the next iteration of `deleteLogs`, which is scheduled sooner if required.

Reviewers: Jun Rao <junrao@gmail.com>, Dong Lin <lindong28@gmail.com>, Ted Yu <yuzhihong@gmail.com>

vahidhashemian committed:
KAFKA-6052; Fix WriteTxnMarkers request retry issue in InterBrokerSendThread (#4705)

This resolves the issue found when running the brokers on Windows which prevents the coordinator from sending WriteTxnMarkers requests to complete a transaction.
